src/app/pages/registration/register-form/register-form.component.html

```diff
@@ -0,0 +1,150 @@
+                  </i> 
+              <input 
+                class="name"
+                formControlName="last_name"
+                placeholder="Last Name" 
+                type='text'
+              />
+              <div *ngIf="registerForm.controls.last_name.invalid && registerForm.controls.last_name.touched" class="alert alert-danger">
+                  <small class="text-danger" *ngIf="registerForm.controls.last_name.errors.required">Last Name is required.</small>
+              </div>
+                </div>
+            </div>
+
+            <div class="input-wrap">
+              <select class="role" formControlName="role"  placeholder=" Select a role">
+                <option value=""  value="null" disabled="true" [selected]="true">Select a role</option>
+                <option value="CA">Client Admin</option>
+                <option value="BY">Buyer</option>
+              </select>
+              <div *ngIf="!registerForm.get('role').valid && registerForm.get('role').touched" class="alert alert-danger">
+                Please select a role
+              </div>
+            
+            </div>
+
+            <div class="input-wrap">
+              
+                <div class="input-icons"> 
+                    <i class="fa fa-lock icon"> 
+                  </i> 
+              <input 
+              class="password"
+              formControlName="password"
+              placeholder="Create Password" 
+              type='password'/>
+
+              <div *ngIf="registerForm.controls.password.invalid && registerForm.controls.password.touched" class="alert alert-danger">
+                  <small class="text-danger" *ngIf="registerForm.controls.password.errors.required">password is required.</small>
+                  <small class="text-danger" *ngIf="registerForm.controls.password.errors.minlength">Password must contain an uppercase, number and a character</small>
+              </div>
+                </div>
+            </div>
+
+
+            <div class="input-wrap">
+                <div class="input-icons"> 
+                    <i class="fa fa-lock icon"> 
+                  </i> 
+              <input
+              class="password"
+              formControlName="confirmed_password"
+              placeholder="Confirm password" 
+              type='password' 
+                 />
+                 <div *ngIf="registerForm.controls.confirmed_password.invalid && registerForm.controls.confirmed_password.touched" class="alert alert-danger">
+                    <small class="text-danger" *ngIf="registerForm.controls.confirmed_password.errors.required">Confirm Password is required.</small>
+                    <div class ="text-danger" *ngIf="registerForm.controls.confirmed_password.errors.mustMatch">Passwords must match</div>
+                </div>
+                </div>
+            </div>
+
+            <button type='submit' [disabled]="!registerForm.valid">Create Account</button>
```

Comment:
Hey @akram256, great work on disabling the submit button when the form is invalid, I realize the button style is the same both when the form is valid and invalid, For better UX we could apply a different style to the button when the form is invalid.we could use style binding to achieve this.